File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Aula 13 10-11-16/fatorialVetor.c
#include <stdio.h>

main()
{
	printf("Vetor de 8 inteiros - fatoriais\n\n");
	
	//vars
	int v[8], res[8], i,j, fat;
	
	//Entradas
	for( i=0; i<8; i++)
	{
		printf("Digite v[%i]: ", i);
		scanf("%d", &v[i]);
	}
	
	//Processamento
	for( i=0; i<8; i++)
	{
		fat = v[i];
		for( j=fat-1; j>1; j--)
		{
			fat *= j;
		}
		res[i] = fat;
	}
	
	//Saída
	printf("\nVetor Principal:\n");
	for( i=0; i<8; i++ )
	{
		printf("%d  ", v[i]);
	}
	
	printf("\nVetor Resultante: \n");
	for( i=0; i<8; i++)
	{
		printf("%d  ", res[i]);
	}
	
}

File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/UNICID - Semestre 2/src/aula11_08_11_16/FatorialVetor.java
//Importando Scanner
import java.util.Scanner;

public class FatorialVetor {

	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner ent = new Scanner(System.in);
		
		System.out.println("Vetor de Fatoriais\n\n");
		
		//vars
		int v[] = new int[8];
		int res[] = new int[8];
		int i, fat, j;
		
		//Entrada
		for( i=0; i<8; i++ )
		{
			System.out.printf("Digite v[%d]: ", i);
			v[i] = ent.nextInt();
		}
		
		//Processamento
		for( i=0; i<8; i++ )
		{
			fat = v[i];
			//Calculando o Fatorial do Número
			for( j=fat-1; j>1; j-- )
			{
				fat*=j;
			}
			res[i] = fat;
		}
		
		//Saída
		System.out.println("\nVetor:");
		for( i=0; i<8; i++)
		{
			System.out.printf("%d ", v[i]);
		}
		
		System.out.println("\nVetor Resultante:");
		for( i=0; i<8; i++ )
		{
			System.out.printf("%d ", res[i]);
		}
		
	}
}
